src/ast/converter.js

```diff
@@ -1199,25 +1199,17 @@ export default class Converter {
     return AConst(StringValue(value));
   }
 
-  GetDate = (date, options, isDateTime) => {
-    date = date || new Date().toISOString();
-
-    if (!isDateTime) {
-      // the `date` value comes in as the string "2017-11-12 23:59:59". We want it to be interpreted as UTC for the
-      // purposes of the SQL query generation. So we convert the local timestamp to a UTC one. We don't care if it's
-      // in a different timezone, we just need to make sure the date component of the timestamp is identical to the
-      // value stored in the date field. We are effectively disregarding the time component of the timestamp.
-      return moment(date.replace(' ', 'T') + 'Z').utc();
-    }
-
+  GetDate = (dateString, options) => {
     const timeZone = (options && options.timeZone) || moment.tz.guess();
 
-    return moment.tz(date, timeZone);
+    return moment.tz(dateString ?? new Date().toISOString(), timeZone);
   }
 
-  ConvertDateValue = (date) => {
+  ConvertDateValue = (expression, date) => {
     if (date) {
-      return date.clone().toISOString();
+      const isTimestamp = expression.column.type !== 'date';
```

Comment:
would it maybe be better to positively check for `date` instead of negatively? if someone naively changed this implementation the logic here could get confusing

Reply:
Good point, something like this may be easier to read:
```js
ConvertDateValue = (expression, date) => {
  if (date) {
    const isDate = expression.column.type === 'date';

    return isDate ? date.format('YYYY-MM-DD') : date.toISOString();
  }
  return null;
}
```

Reply:
Good call, I just added a helper method for completeness with the other helper methods that already exist.

Reply:
thanks @zhm 💪